array_api_tests/test_constants.py

```diff
@@ -1,34 +1,51 @@
-from ._array_module import (e, inf, nan, pi, equal, isnan, abs, full,
-                            float64, less, isinf, greater, all)
-from .array_helpers import one
+import math
+from typing import Any, SupportsFloat
 
-def test_e():
-    # Check that e acts as a scalar
-    E = full((1,), e, dtype=float64)
+import pytest
 
-    # We don't require any accuracy. This is just a smoke test to check that
-    # 'e' is actually the constant e.
-    assert all(less(abs(E - 2.71), one((1,), dtype=float64))), "e is not the constant e"
+from . import dtype_helpers as dh
+from ._array_module import mod as xp
+from .typing import Array
 
-def test_pi():
-    # Check that pi acts as a scalar
-    PI = full((1,), pi, dtype=float64)
 
-    # We don't require any accuracy. This is just a smoke test to check that
-    # 'pi' is actually the constant π.
-    assert all(less(abs(PI - 3.14), one((1,), dtype=float64))), "pi is not the constant π"
+def assert_scalar_float(name: str, c: Any):
+    assert isinstance(c, SupportsFloat), f"{name}={c!r} does not look like a float"
+
+
+def assert_0d_float(name: str, x: Array):
+    assert dh.is_float_dtype(
+        x.dtype
+    ), f"xp.asarray(xp.{name})={x!r}, but should have float dtype"
+
+
+@pytest.mark.parametrize("name, n", [("e", math.e), ("pi", math.pi)])
+def test_irrational(name, n):
```

Comment:
This test is actually testing that n is fractional, not irrational (floating point numbers aren't strictly "irrational", and it's impossible to test if they "represent" one).

Reply:
I guess the "irrational" here means it's testing the irrational numbers. A bit confusing. Maybe `test_irrational_numbers` would be clearer.